Migration spec prune

kbrock opened the pull request:
Please review #3355 first

We do not copy schema only nor do we import or export to files in our migrations.

/cc @Fryguy

Fryguy commented:
I would prefer to keep these utility functions.  Perhaps we can move them into a MigrationHelper::BulkCopier or something?  They have been used in the past when we need it for very large datasets.

I do agree that we can drop all the database redirection though.  At this point we are only working with postgresql and all that is a holdover from when we supported sql server as well.

matthewd commented:
@Fryguy this seems simple enough to me that we can just relegate it to git history; if we have need of it again, we shouldn't have any problem getting it back... and it doesn't even seem to really be doing all that much to recreate.

Alternatively, maybe we can drop it into a gist or something? I'm not sure exactly where, but ISTM we'd benefit from having this live somewhere that marks it as "historical curiosity, which may be a useful starting point in the future", and distinguishes it from actual-code-we-need-to-maintain.

Fryguy commented:
It took me a while to write it in the first place, so I'd argue against the "simple enough" part :smile:

However, I agree that if we have a good commit message it should be easy enough to find in the git history.  So, I'd be up for removing it if it weren't using it.  I agree I can gist it for my own purposes if I need to (that is, extracting our pg enhancements to our own gem or something like pghero)

BUT we *are* using it.  @kbrock is trying to remove it in #3355, but it's not out.  @kbrock Can you please mark this as dependent on that...or change this one to just remove the pieces we really aren't using?

kbrock commented:
@Fryguy Ooh. good catch. thanks.

I split out a single commit into multiple PRs and didn't realize I had removed the line that depended upon this.

kbrock commented:
If this is dependent. Then this should blow up the migration specs. I'm not so sure this is used.

matthewd commented:
Yeah, I think the thing changed in #3355 is using the still-present (and remarkably similar) [`bulk_copy_transfer`](https://github.com/ManageIQ/manageiq/pull/3356/files#diff-5c22c5eb4fa33f2189cc9fdbabe68f8dR297)

Fryguy commented:
> Then this should blow up the migration specs.

You give me way too much credit :laughing:

chessbyte commented:
@kbrock Can you extract the first commit labeled `remove meta around migration bulk copy helpers` to a separate PR since that is non-controversial.  The other 2 commits in this PR remove methods that may or may not be used and/or needed in the future, causing the controversy.

Fryguy commented:
Oh I see what you mean @matthewd .

so it seems `bulk_copy_*` was for "export to file, then import from file", and `bulk_copy_transfer` is for `export to STDOUT and pipe into import via STDIN`.  The latter is better because it doesn't blow out disk space and is what's used by the `copy_table_via_bulk_copy` method, so yeah this is all fine.  I didn't see that in the diff, so it confused me.
